tools/Vrops.py

```diff
@@ -149,368 +124,120 @@ def get_resources(self, target, token, uuids, resourcekinds):
         headers = {
             'Content-Type': "application/json",
             'Accept': "application/json",
-            'Authorization': "vRealizeOpsToken " + token
+            'Authorization': f"vRealizeOpsToken {token}"
         }
-        import queue
+
         q = queue.Queue()
         thread_list = list()
         chunk_iteration = 0
 
+        logger.debug('>----------------------- get_latest_values_multiple')
+        logger.debug(f'target   : {target}')
+        logger.debug(f'collector: {collector}')
+        logger.debug(f'Amount keys : {len(keys)}')
+        for k in keys:
+            logger.debug(f'key  : {k}')
+
         for uuid_list in uuids_chunked:
             chunk_iteration += 1
-            t = Thread(target=Vrops.get_project_id_chunk,
-                       args=(q, uuid_list, url, headers, target, chunk_iteration))
+            t = Thread(target=Vrops._get_chunk,
+                       args=(q, uuid_list, url, headers, keys, target, kind, collector, chunk_iteration))
             thread_list.append(t)
             t.start()
         for t in thread_list:
             t.join()
 
+        return_list = list()
+        response_status_code = 503
+
         while not q.empty():
-            project_ids += q.get()
+            returned_chunks = q.get()
+            response_status_code = returned_chunks[1] if returned_chunks[1] > 200 else 200
+            return_list.extend(returned_chunks[0])
 
         logger.debug(f'Amount uuids: {len(uuids)}')
-        logger.debug(f'Fetched     : {len(project_ids)}')
+        logger.debug(f'Fetched     : {len({r.get("resourceId") for r in return_list})}')
         logger.debug('<--------------------------------------------------')
 
-        return project_ids
+        return return_list, response_status_code
 
-    # only recommended for a small number of statkeys and resources.
-    def get_latest_stat(target, token, uuid, key, collector):
-        logger.debug('>---------------------------------- get_latest_stat')
-        logger.debug(f'key      : {key}')
-        logger.debug(f'target   : {target}')
-        logger.debug(f'collector: {collector}')
+    def get_latest_properties_multiple(self, target: str, token: str, uuids: list, keys: list, collector: str):
+        return self.get_latest_values_multiple(target, token, uuids, keys, collector, kind='properties')
 
-        url = "https://" + target + "/suite-api/api/resources/" + uuid + "/stats/latest"
-        headers = {
-            'Content-Type': "application/json",
-            'Accept': "application/json",
-            'Authorization': "vRealizeOpsToken " + token
-        }
-        disable_warnings(exceptions.InsecureRequestWarning)
+    def get_latest_stats_multiple(self, target: str, token: str, uuids: list, keys: list, collector: str):
+        return self.get_latest_values_multiple(target, token, uuids, keys, collector, kind='stats')
```

Comment:
it is really cool, that stats and properties are leveraging the same functionality now